pkg/webreg/webreg.go

```diff
@@ -365,6 +366,253 @@ const templateDefinitions = `
+For instance:
+<p>
+
+<code>Dockerfile</code>:
+{{ dockerfileSyntax (index . "multistageDockerfile") }}
+
+<p>
+While such a <code>Dockerfile</code> could simply be built by <code>ci-operator</code>,
+a number of optimizations can be configured to speed up the process, especially if
+multiple output images share artifacts. An output container image build is configured
+for <code>ci-operator</code> with the <code>images</code> stanza in the configuration.
+Any entry in the <code>images</code> stanza can be configured with native OpenShift
+<code>Builds</code> options; the full list can be viewed <a href="https://godoc.org/github.com/openshift/ci-tools/pkg/api#ProjectDirectoryImageBuildInputs">here.</a>
+In the following example, an output container image is built where the <code>builder</code>
+image is replaced with the image layers containing built artifacts in <code>pipeline:bin</code>
+and the output image base is replaced with the appropriate entry from <code>base_images</code>.
+<p>
+
+<code>ci-operator</code> configuration:
+{{ yamlSyntax (index . "ciOperatorImageConfig") }}
+
+<p>
+By making use of the previously compiled artifacts in the intermediate <code>pipeline:bin</code>
+image, this repository is able to cache the Go build. If multiple output images exist that
+rely on built artifact, this caching effect can reduce build times dramatically.
+</p>
+
+<h4 id="promotion"><a href="#promotion">Publishing Container Images</a></h4>
+
+<p>
+Once <code>ci-operator</code> has build output container images for a repository,
+it can furthermore publish them to an integration <code>ImageStream</code> so that
+other repositories can consume them. For instance, every image that makes up the
+OpenShift release payload is incrementally updated in an integration <code>ImageStream</code>
+so that release payloads may be created from the latest tested version of every
+component. In order to publish images, add the <code>promotion</code> stanza to
+<code>ci-operator</code> configuration.
+</p>
+
+<p>
+The <code>promotion</code> stanza declares which container images are published
+and defines the integration <code>ImageStream</code> where they will be available.
+By default, all container images declared in the <code>images</code> block of a
+<code>ci-operator</code> configuration are published when a <code>promotion</code>
+stanza is present to define the integration <code>ImageStream</code>. Promotion can
+be furthermore configured to include other images by setting <code>additional_images</code>
+and to exclude images included by default with <code>excluded_images</code>. The
+following example publishes the following images:
+</p>
+<ul>
+  <li>the <code>pipeline:src</code> tag, published as <code>ocp/4.5:repo-scripts</code> containing the latest version of the repository</li>
+  <li>the <code>stable:component</code> tag, published as <code>ocp/4.5:mycomponent</code> containing the output component itself</li>
```

Comment:
it's not clear where stable:component comes from in the example since you didn't include the images stanza

Reply:
it's also not clear what the exclude is applied to since the example doesn't include the images stanza that produced the `mytests` image

Reply:
How can I better explain that all the different `ci-operator` configuration snippets are part of one larger one? I only in-line one bit at a time.

Reply:
inline more?

```
promotion stanza
....
additional context
```

Reply:
I mean every single inline section only shows one bit and I kind of want to keep it that way, it helps to focus the doc.

Reply:
i think you can teach more about the section you're focusing on if you include the context of other sections it relates to/draws from.  but it's your doc.